File: src/main/java/com/dh/ondot/member/domain/enums/AddressType.java
package com.dh.ondot.member.domain.enums;

import com.dh.ondot.core.exception.UnsupportedException;

import java.util.Locale;

import static com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_ADDRESS_TYPE;

public enum AddressType {
    HOME;

    public static AddressType from(String type) {
        try {
            return AddressType.valueOf(type.toUpperCase(Locale.ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException(UNSUPPORTED_ADDRESS_TYPE, type);
        }
    }
}

File: src/main/kotlin/com/dh/ondot/member/domain/enums/AddressType.kt
package com.dh.ondot.member.domain.enums

import com.dh.ondot.core.exception.ErrorCode.UNSUPPORTED_ADDRESS_TYPE
import com.dh.ondot.core.exception.UnsupportedException

enum class AddressType {
    HOME,
    ;

    companion object {
        @JvmStatic
        fun from(type: String): AddressType =
            entries.find { it.name.equals(type, ignoreCase = true) }
                ?: throw UnsupportedException(UNSUPPORTED_ADDRESS_TYPE, type)
    }
}
